src/Report/EnabledMembers.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace SilverStripe\MFA\Report;
+
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Forms\CheckboxField;
+use SilverStripe\Forms\DropdownField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\NumericField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\MFA\Method\MethodInterface;
+use SilverStripe\MFA\Service\MethodRegistry;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\Reports\Report;
+use SilverStripe\Security\Member;
+
+if (!class_exists(Report::class)) {
+    return;
+}
+
+class EnabledMembers extends Report
+{
+    protected $title = 'Members with MFA enabled on their account';
+
+    protected $description = 'Identifies all members on this site, with columns added to indicate the status each has'
+        . ' in regards to multi factor authentication method registration. E.g. how many methods are set up by that'
+        . ' member, which method types, and who has opted not to set up MFA (if this option is enabled for the site)';
+
+    protected $dataClass = Member::class;
+
+    public function title()
+    {
+        return _t(__CLASS__ . '.TITLE', parent::title());
+    }
+
+    public function description()
+    {
+        return _t(__CLASS__ . '.DESCRIPTION', parent::description());
+    }
+
+    /**
+     * @param array $params
+     */
+    public function sourceRecords($params)
+    {
+        $methods = $this->getMethodClassToTitleMapping();
+
+        // Having a WHERE limits the results, which means the the COUNT will be off (GROUP BY applied after WHERE)
+        // So clip the count filter and apply it after the database query, only if Methods filter is also set
+        // this way we can delegate to the database where possible, which should be faster.
+        if (isset($params['Count'])) {
+            $desiredCount = (int)$params['Count'];
+            if (isset($params['Methods'])) {
+                $countFilter = $desiredCount;
+                unset($params['Count']);
+            } elseif ($desiredCount) {
+                $params['Count'] = $desiredCount++;
+            }
+        }
+
+        $members = $this->applyParams(Member::get(), $params)->toArray();
+        array_walk($members, function ($member) use ($methods) {
```

Comment:
You can typehint member here:

```suggestion
        array_walk($members, function (Member $member) use ($methods) {
```

Additionally, I think this is an example of an `array_walk` that should just be a `foreach`.

Reply:
It was refactored :)